packages/components/src/components/form/Select/index.tsx

```diff
@@ -306,78 +278,83 @@ const Select = ({
 
         // returns "undefined" if no option was found
         return foundOption;
-    };
-
-    const scrollToOption = (option: Option) => {
-        if (selectRef.current) {
-            // I found a way how to scroll on and option in this tutorial: https://github.com/JedWatson/react-select/issues/3648
-            selectRef.current.select.scrollToFocusedOptionOnUpdate = true;
-            selectRef.current.select.setState({
-                focusedValue: null,
-                focusedOption: option,
-            });
-        }
-    };
-
-    const onKeyDown = async (event: React.KeyboardEvent) => {
-        // This function is executed when user presses keyboard
-        if (useKeyPressScroll) {
-            // Get value of the key pressed by the user
-            const charValue = event.key;
-
-            // Get current timestamp and check how long it is since the last keyPress event happened.
-            const currentTimestamp = new Date().getTime();
-            const timeSincePreviousKeyPress = currentTimestamp - lastKeyPressTimestamp.current;
-
-            // Save current timestamp to lastKeyPressTime variable
-            lastKeyPressTimestamp.current = currentTimestamp;
-
-            // If user didn't type anything for 0.8 seconds, set searchedValue to just pressed key, otherwise add the new value to the old one
-            if (timeSincePreviousKeyPress > 800) {
-                searchedTerm.current = charValue;
-            } else {
-                searchedTerm.current += charValue;
-            }
+    }, []);
 
+    const scrollToOption = useCallback(
+        (option: Option) => {
             if (selectRef.current) {
-                // Get options object
-                const { options } = selectRef.current.select.props;
+                // I found a way how to scroll on and option in this tutorial: https://github.com/JedWatson/react-select/issues/3648
+                selectRef.current.select.scrollToFocusedOptionOnUpdate = true;
+                selectRef.current.select.setState({
+                    focusedValue: null,
+                    focusedOption: option,
+                });
+            }
+        },
+        [selectRef],
+    );
 
-                if (options && options.length > 1) {
-                    /*
+    const onKeyDown = useCallback(
+        async (event: React.KeyboardEvent) => {
+            // This function is executed when user presses keyboard
```

Comment:
please try to clean comments here, most of them is really useless since they are explaining what is obvious

Reply:
done